dataframe_show_reader/dataframe_assert_equal.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 The DataFrame Show Reader Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyspark.sql import DataFrame
+from pyspark.sql.types import StructField, StructType
+
+
+def assert_equal(expected: DataFrame, actual: DataFrame,
+                 show_output_on_failure: bool = True):
+    """Assert that two DataFrames contain the same data.
+    :param expected: The expected DataFrame
+    :param actual: The actual DataFrame
+    :param show_output_on_failure: If the DataFrames are not equal, show
+    the DataFrame schema or data (depending on where the mismatch is) in order
+    to help debugging.
+    :return: None
+    """
+    if expected is None:
+        assert actual is None, \
```

Comment:
I'm thinking that the error message (`The [x] DataFrame is None, but the [y] DataFrame is not.`) contains the important information and that the show output would be a distraction, but I could also be convinced that a strict literal interpretation of `show_output_on_failure` is appropriate. Although, if there is a schema mismatch, we call `printSchema` rather than `show`, which I think is the most useful output there, so maybe we should change the parameter name to `verbose`. I think I favor changing the parameter name to `verbose` and leaving the rest the same. Any thoughts?

Reply:
Yes, i like the `verbose` suggestion and leaving it as is.